src/include/migraphx/operators.hpp

```diff
@@ -631,6 +633,56 @@ struct as_shape
     int output_alias(const std::vector<shape>&) const { return 0; }
 };
 
+struct gather
+{
+    std::size_t axis = 0;
+    std::string name() const { return "gather"; }
+
+    shape compute_shape(std::vector<shape> inputs) const
+    {
+        check_shapes{inputs, *this}.has(2);
+        auto lens = inputs[0].lens();
+        if(axis >= lens.size())
+        {
+            MIGRAPHX_THROW("Gather, axis is out of range.");
+        }
+        auto type  = inputs[0].type();
+        lens[axis] = inputs[1].elements();
+
+        return {type, lens};
+    }
+
+    template <class T>
+    void compute_index(const T& out_idx, const std::vector<argument>& args, T& in_idx) const
+    {
+        in_idx = out_idx;
+        // max dimension in axis
+        std::size_t max_dim = args[0].get_shape().lens()[axis];
+        std::size_t idx     = args[1].at<std::size_t>(out_idx[axis]);
```

Comment:
We should convert `args[1]` to `std::vector<std::size_t>` first, instead of calling `.at`. This is because `.at` uses a switch statement and its repeatedly called inside the loop, which will be rather slow.

Reply:
Fixed